---
title: "05_analysis_1"
format:
  html:
    embed-resources: true
editor: visual
---

## **Loading libraries**

```{r}
library("tidyverse")
library("dplyr")
library("here")
```

## **Loading data**

Loading the data from the previous augmentation step *03_augment*.

```{r}
data_metrics <- read_tsv(here("data","data_metrics_aug.tsv"))
data_gene <- read_tsv(here("data","data_gene_aug.tsv"))
data_score <- read_tsv(here("data","data_score_aug.tsv"))
data_code <- read_tsv(here("data","data_code_aug.tsv"))
data_diag <- read_tsv(here("data","data_diag_aug.tsv"))
```

-   data_metrics: Contains clinical and demographic metadata for patients, including diagnosis age, sex, genetic ancestry, race, and ethnicity. This dataset provides socio-demographic context and basic patient characteristics important for stratification and subgroup analyses.

-   data_gene: Contains genomic alteration metrics per patient, such as fraction of the genome altered, mutation counts, tumor mutation burden (TMB), and tumor break load. This data captures key genomic instability features relevant to tumor biology and mutation burden.

-   data_score: Contains various biological and clinical scores and codes, including aneuploidy score, multiple hypoxia-related scores, microsatellite instability (MSI) scores from different methods, and tissue source site identifiers. These functional scores inform on genomic instability, tumor microenvironment, and sample origin.

-   data_code: Contains standardized oncology disease stage and classification codes per patient. This includes AJCC cancer staging codes, ICD-O-3 histology and site codes, and codes for metastasis and lymph node staging. This set provides clinical staging and diagnostic classification for tumors.

-   data_diag: Contains detailed clinical diagnosis and outcome variables such as tumor type, overall survival time and status, disease-specific survival, disease-free survival, progression-free survival, prior diagnoses, and therapy details. Also includes cancer status indicators, neoplasm histologic grade, and ICD-10 classification. This dataset focuses on clinical endpoints and diagnostic outcomes.

## Plotting

### Demographic & Clinical Metadata (data_metrics) 

```{r}
data_metrics
```

Needs to go to data_aug this!

```{r}
data_metrics <- data_metrics |>
  mutate(Age_Group = cut(
    `Diagnosis Age`,
    breaks = seq(20, 90, by = 10),
    right = FALSE))
```

```{r}
plot_data <- data_metrics |> 
  count(Sex, Age_Group)

sex_number <- data_metrics |> 
  count(Sex) |> 
  mutate(Age_Group = "Total")

# Bind rows
plot_data_total <- bind_rows(plot_data, sex_number)

# We can try to add a dummy row to separate it from the others
# I don´t know if it´s a freat ideia to plot the Total in there, it was just an ideia

# Now plot, showing "Total" along x-axis

plot_data_total |> 
  ggplot(mapping = 
           aes(x = Sex,
               y = n,
               fill = Age_Group)) +
  geom_bar(stat = "identity", position = position_dodge(preserve = "single"), color = "black", alpha = 0.5)
```

```{r}
data_metrics |> 
  ggplot(mapping = 
           aes(x = `Diagnosis Age`,
               fill = Sex)) +
  geom_density(alpha = 0.3)
```

Race, Genetic Ancestry Label, and Ethnicity Category strongly overlap in our data, with most individuals of European ancestry also classified as White and similar by ethnicity. This means we lack sufficient diversity to draw meaningful conclusions from subgroup analyses of these categories.

```{r}
data_metrics |> 
  ggplot(mapping = 
           aes(x = `Race Category`,
               fill = `Genetic Ancestry Label`)) +
  geom_bar(position = position_dodge(preserve = "single"), color = "black", alpha = 0.7) +
  geom_vline(xintercept = seq(0.5, length(unique(data_metrics$`Race Category`)) + 0.5, by = 1), linetype = "dashed") +
  theme_minimal()
```

### Genomic Alterations & Instability (data_gene) 

```{r}
data_metrics |> 
  select(-c(`Genetic Ancestry Label`, `Race Category`, `Ethnicity Category`))

data_metrics_gene <- full_join( data_metrics, data_gene, by="Patient ID")

data_metrics_gene
```

Fraction Genome Altered - The proportion of the tumor’s genome with copy number gains or losses, reflecting the extent of chromosomal instability present in the cancer cells.

Mutation Count - The total number of mutations identified in the tumor genome, typically including single-nucleotide variants and small insertions/deletions.

TMB (nonsynonymous) - Tumor Mutation Burden (TMB) is the number of nonsynonymous mutations (those that change protein sequences) per megabase of DNA in the tumor, and is used to estimate overall mutational load.

Tumor Break Load - The quantity or extent of structural variants in the tumor genome, representing breaks or major rearrangements, which indicate higher genomic instability in the tumor.

```{r}
library(corrplot)
cor_matrix <- cor(data_metrics_gene |>
                    select(`Fraction Genome Altered`, 
                           `Mutation Count`, 
                           `TMB (nonsynonymous)`, 
                           `Tumor Break Load`), 
                    use = "complete.obs")

corrplot(cor_matrix, method = "color")
```

------------------------------------------------------------------------

```{r}
data_metrics_gene |>
  filter(`Mutation Count` < 5000)
```

Before plotting, that one strange outlier should be removed to avoid skewing the results. The resulting trend makes sense, since mutations occur randomly across the genome, the proportion that are nonsynonymous tends to follow a predictable pattern, assuming random mutagenesis within tumor DNA and consistent coding sequence proportions.

```{r}
data_metrics_gene |> 
  filter(`TMB (nonsynonymous)` < 400) |> 
  ggplot(mapping = 
           aes(x = `Tumor Break Load`,
               y = `TMB (nonsynonymous)`,
               color = Age_Group)) +
  geom_point()
```

```{r}
data_metrics_gene |> 
  filter(`TMB (nonsynonymous)` < 400) |> 
  ggplot(mapping = 
           aes(x = `Fraction Genome Altered`,
               y = `Tumor Break Load`,
               color = Age_Group)) +
  geom_point()
```

### Box Plots of data_gene on certain groups to uncover whether certain groups harbor more genomic instability.

```{r}
data_code |>
  count(
    `Neoplasm Disease Stage American Joint Committee on Cancer Code`,
    `International Classification of Diseases for Oncology, Third Edition ICD-O-3 Histology Code`
  ) |> 
  ggplot(
    mapping = aes(
  x = `Neoplasm Disease Stage American Joint Committee on Cancer Code`,
  y = `International Classification of Diseases for Oncology, Third Edition ICD-O-3 Histology Code`,
  fill = n)) +
  geom_tile() +
  theme(axis.text.x = element_text(angle=45, hjust=1)) +
  labs(fill = "Count")
```

```{r}
data_gene_code = full_join(data_gene, data_code, by = "Patient ID")
```

```{r}
data_gene_code |>
  filter(`Mutation Count` < 5000) |> 
  pivot_longer(cols = c(`Fraction Genome Altered`, 
                        `Mutation Count`, 
                        `TMB (nonsynonymous)`, 
                        `Tumor Break Load`),
               names_to = "Genomic_Metric", 
               values_to = "Value") |> 
ggplot(mapping = 
         aes(x = `Neoplasm Disease Stage American Joint Committee on Cancer Code`, 
             y = Value)) +
  geom_boxplot() +
  facet_wrap(~ Genomic_Metric, scales = "free_y")
```

### Different Classification Methods Coherence

```{r}
data_code
```

```{r}
data_code |>
  count(
    `Neoplasm Disease Stage American Joint Committee on Cancer Code`,
    `International Classification of Diseases for Oncology, Third Edition ICD-O-3 Histology Code`
  ) |> 
  ggplot(
    mapping = aes(
      x = `Neoplasm Disease Stage American Joint Committee on Cancer Code`,
      y = `International Classification of Diseases for Oncology, Third Edition ICD-O-3 Histology Code`,
      fill = n)
  ) +
  geom_tile() +
  geom_text(aes(label = n), color = "white") +
  theme(axis.text.x = element_text(angle=45, hjust=1)) +
  labs(fill = "Count")
```

```{r}
data_code |>
  count(
    `Neoplasm Disease Stage American Joint Committee on Cancer Code`,
    `International Classification of Diseases for Oncology, Third Edition ICD-O-3 Histology Code`
    
  ) |> 
  ggplot(
    mapping = aes(
  x = `Neoplasm Disease Stage American Joint Committee on Cancer Code`,
  y = `International Classification of Diseases for Oncology, Third Edition ICD-O-3 Histology Code`,
  fill = n)) +
  geom_tile() +
  theme(axis.text.x = element_text(angle=45, hjust=1)) +
  labs(fill = "Count")
```

### Bin

```{r}
working_df |> 
  ggplot(mapping =
           aes(x= `Age_Group`, 
               y = `Fraction Genome Altered`,
               color = Sex)) +
  geom_boxplot()
```

```{r}
working_df |> 
  ggplot(mapping = 
           aes( x = `Disease Free (days)`,
                color = Sex)) +
  geom_density()
```

```{r}
#| fig-height: 
working_df |> 
  ggplot(mapping = 
           aes( x = `Tumor Type`,
                fill = `Overall Survival Status (des)`)) +
  geom_bar(position = position_dodge(preserve = "single"), color = "black", alpha = 0.5) +
  theme(
    panel.grid.major.x =  element_blank(),
    axis.text.x = element_text(angle = 70, 
                               hjust = 1,
                               vjust = 1)
  )
```
