---
title: "05_analysis_2"
format: html
---

## **Loading libraries**

```{r}
library(patchwork)
library(tidyverse)
library(ggplot2)

library(survival)
library(survminer)
```

## **Loading data**

Loading the data from the previous augmentation step *03_augment*.

```{r}
#| message: false
data_metrics <- read_tsv(here("data","data_metrics_aug.tsv"))
data_gene <- read_tsv(here("data","data_gene_aug.tsv"))
data_score <- read_tsv(here("data","data_score_aug.tsv"))
data_code <- read_tsv(here("data","data_code_aug.tsv"))
data_diag <- read_tsv(here("data","data_diag_aug.tsv"))
```

Previous analysis from André (just for context):

-   **data_metrics** — patient demographics (age, sex, race, ancestry)
-   **data_gene** — genomic instability metrics (mutation count, TMB, fraction genome altered).
-   **data_score** — biological scores (aneuploidy, hypoxia, MSI).
-   **data_code** — cancer staging and clinical classification codes.
-   **data_diag** — diagnoses, survival outcomes, therapy history.

Important context from before:

-   Merged `data_metrics` and `data_gene` → `data_metrics_gene`. Strong correlations between 4 genomic instability metrics (TMB, Mutation Count, Tumor Break Load, Fraction Genome Altered)

-   **Plot genomic metrics vs age group, tumor break load, and fraction genome altered –\>** Higher genomic instability measures tend to cluster together

-   Joined `data_gene` + `data_code` (cancer stage classifications)→ data_gene_code. Conclusions: ??? confusion

### Next Analysis (in points)

#### Genomic alterations and instability (data_gene)

-   **Comparing survival time between high and low** genomic instability measures / Relationship between genomic instability and survival status

Comment André -\> i don´t know if you can´t do this in tydiverse

```{r}
#| fig-width: 10
#| fig-height: 7

data_gene_diag <- full_join(data_gene, data_diag, by = "Patient ID")
data_gene_diag <- data_gene_diag |>
  filter(`Mutation Count` < 5000) |>
  filter(`TMB (nonsynonymous)` < 400)
  
# Survival days vs TMB
tmb_med <- median(data_gene_diag[["TMB (nonsynonymous)"]], na.rm = TRUE)  

data_gene_diag <- data_gene_diag |>
  mutate(
    TMB_Group = case_when(
      `TMB (nonsynonymous)` > tmb_med ~ "High TMB",
      `TMB (nonsynonymous)` <= tmb_med ~ "Low TMB"
    ))

plt1 <- ggplot(data_gene_diag, 
               aes(x = `Overall Survival (days)`, 
                   y = TMB_Group, 
                   fill = TMB_Group)) +
  geom_violin(alpha = 0.6) +
  geom_jitter(width = 0.2, alpha = 0.6) +
  theme_minimal() +
  ggtitle("Overall Survival vs TMB")

plt2 <- data_gene_diag |>
  ggplot(aes(y = `TMB (nonsynonymous)`, 
             x = `Overall Survival (days)`,
             color = TMB_Group)) +
  geom_point(alpha = 0.6) +
  geom_smooth(method = "loess", color = "blue") +
  theme_minimal() +
  ggtitle("Overall Survival vs TMB")

plt1 / plt2

# Survival days vs Mutation Count
mutcount_med <- median(data_gene_diag[["Mutation Count"]], na.rm = TRUE)

data_gene_diag <- data_gene_diag |>
  mutate(
    MutCount_Group = case_when(
      `Mutation Count` > mutcount_med ~ "High Mut Count",
      `Mutation Count` <= mutcount_med ~ "Low Mut Count"
    ))

plt3 <- ggplot(data_gene_diag, 
               aes(x = `Overall Survival (days)`, 
                   y = MutCount_Group, 
                   fill = MutCount_Group)) +
  geom_violin(alpha = 0.6) +
  geom_jitter(width = 0.2, alpha = 0.6) +
  theme_minimal() +
  ggtitle("Overall Survival vs Mutation Count")

plt4 <- data_gene_diag |>
  ggplot(aes(y = `Mutation Count`, 
             x = `Overall Survival (days)`,
             color = MutCount_Group)) +
  geom_point(alpha = 0.6) +
  geom_smooth(method = "loess", color = "blue") +
  theme_minimal() +
  ggtitle("Overall Survival vs Mutation Count")

plt3 / plt4

# Survival days vs Fraction Genome Altered
frac_med <- median(data_gene_diag[["Fraction Genome Altered"]], na.rm = TRUE)

data_gene_diag <- data_gene_diag |>
  mutate(
    Frac_Group = case_when(
      `Fraction Genome Altered` > frac_med ~ "High Fraction Genome Altered",
      `Fraction Genome Altered` <= frac_med ~ "Low Fraction Genome Altered"
    ))

plt5 <- ggplot(data_gene_diag, 
               aes(x = `Overall Survival (days)`, 
                   y = Frac_Group, 
                   fill = Frac_Group)) +
  geom_violin(alpha = 0.6) +
  geom_jitter(width = 0.2, alpha = 0.6) +
  theme_minimal() +
  ggtitle("Overall Survival vs Fraction Genome Altered")

plt6 <- data_gene_diag |>
  ggplot(aes(y = `Fraction Genome Altered`, 
             x = `Overall Survival (days)`,
             color = Frac_Group)) +
  geom_point(alpha = 0.6) +
  geom_smooth(method = "loess", color = "blue") +
  theme_minimal() +
  ggtitle("Overall Survival vs Fraction Genome Altered")

plt5 / plt6

# Survival days vs Tumor Break Load
tbload_med <- median(data_gene_diag[["Tumor Break Load"]], na.rm = TRUE)

data_gene_diag <- data_gene_diag |>
  mutate(
    TumorLoad_Group = case_when(
      `Tumor Break Load` > tbload_med ~ "High Tumor Break Load",
      `Tumor Break Load` <= tbload_med ~ "Low Tumor Break Load"
    )
  )

plt7 <- ggplot(data_gene_diag, 
               aes(x = `Overall Survival (days)`, 
                   y = TumorLoad_Group, 
                   fill = TumorLoad_Group)) +
  geom_violin(alpha = 0.6) +
  geom_jitter(width = 0.2, alpha = 0.6) +
  theme_minimal() +
  ggtitle("Overall Survival vs Tumor Break Load")

plt8 <- data_gene_diag |>
  ggplot(aes(y = `Tumor Break Load`, 
             x = `Overall Survival (days)`,
             color = TumorLoad_Group)) +
  geom_point(alpha = 0.6) +
  geom_smooth(method = "loess", color = "blue") +
  theme_minimal() +
  ggtitle("Overall Survival vs Tumor Break Load")

plt7 / plt8
```

-   **Tumor types and Genomic differences across them (not meaningful for further analysis)**

```{r}
#| fig-width: 8
#| fig-height: 5

data_gene_diag |>
  filter(
    `Tumor Type` %in% c(
      "Pancreas Adenocarcinoma, Ductal Type",
      "Pancreas Adenocarcinoma, Other Subtype"
    ),
    `Mutation Count` < 5000,
    `TMB (nonsynonymous)` < 400
  ) |>
  pivot_longer(
    cols = c(
      `Mutation Count`,
      `TMB (nonsynonymous)`,
      `Fraction Genome Altered`,
      `Tumor Break Load`
    ),
    names_to = "Metric",
    values_to = "Value"
  ) |>
  ggplot(aes(x = `Tumor Type`, y = Value, fill = `Tumor Type`)) +
  geom_boxplot() +
  facet_wrap(~ Metric, ncol = 2, scales = "free_y") +
  theme_minimal() +
  theme(
    axis.title.x = element_blank(),
    axis.text.x  = element_blank(),
    axis.ticks.x = element_blank()
  )


```

[Conclusion]{.underline}: Other subtypes of adenocarcinoma have a higher fraction of the genome altered compared to others. From now on discart all tumor types that are not pancreas adenocarcinoma (ductal type or other subtype)

-   Show similarities in **tumor type** + stage + genomic instability measures

    ```{r}
    #| fig.width=10
    #| fig.height=6


    # Merge data_gene, data_code, and data_diag
    data_full <- data_gene |>
      full_join(data_code, by = "Patient ID") |>
      full_join(data_diag, by = "Patient ID")

    data_full <- data_full |>
      rename(`Cancer Stage` = `Neoplasm Disease Stage American Joint Committee on Cancer Code`) |>
      filter(`Mutation Count` < 5000) |>
      filter(`TMB (nonsynonymous)` < 400) |>
      filter(`Tumor Type` == "Pancreas Adenocarcinoma, Ductal Type" |
        `Tumor Type` == "Pancreas Adenocarcinoma, Other Subtype")


    #### Just tumor type first
    # Stage distribution per tumor type
    plt_stage_dist <- data_full |>
      ggplot(aes(x = `Cancer Stage`, fill = `Cancer Stage`)) +
      geom_bar() +
      facet_wrap(~ `Tumor Type`, scales = "free_y") +
      theme_minimal() +
      ggtitle("Cancer Stage Distribution by Tumor Type") +
      theme(axis.text.x = element_text(angle = 45, 
                                       hjust = 1),
            legend.position = "none") +
      labs(x = "Cancer Stage", y = "Count")


    # Mutation Count by Tumor Type and Stage
    plt_mut_stage <- data_full |>
      ggplot(aes(x = `Cancer Stage`, 
                 y = `Mutation Count`, 
                 fill = `Cancer Stage`)) +
      geom_boxplot() +
      theme_minimal() +
      ggtitle("Mutation Count by Tumor Type and Stage") +
      theme(axis.text.x = element_text(
        angle = 45,
        hjust = 1)) +
      facet_wrap(~ `Tumor Type`, scales = "free_x")

    # Fraction Genome Altered by Tumor Type and Stage
    plt_frac_stage <- data_full |>
      ggplot(aes(x = `Cancer Stage`, 
                 y = `Fraction Genome Altered`, 
                 fill = `Cancer Stage`)) +
      geom_boxplot() +
      theme_minimal() +
      ggtitle("Fraction Genome Altered by Tumor Type and Stage") +
      theme(axis.text.x = element_text(
        angle = 45, 
        hjust = 1)) +
      facet_wrap(~ `Tumor Type`, scales = "free_x")

    # Tumor Break Load by Tumor Type and Stage
    plt_tbload_stage <- data_full |>
      ggplot(aes(x = `Cancer Stage`, 
                 y = `Tumor Break Load`, 
                 fill = `Cancer Stage`)) +
      geom_boxplot() +
      theme_minimal() +
      ggtitle("Tumor Break Load by Tumor Type and Stage") +
      theme(axis.text.x = element_text(
        angle = 45, 
        hjust = 1)) +
      facet_wrap(~ `Tumor Type`, scales = "free_x")

    # Combine plots
    plt_stage_dist 
    plt_mut_stage 
    plt_frac_stage 
    plt_tbload_stage


    ```

-   One more test with tumor_type

#### Tests with the tumor scores

IDEAAA: comparing all together the **genomic instability measures** (TMB, Mutation Count, Tumor Break Load, Fraction Genome Altered) in `data_gene`; the **biological scores** ("Aneuploidy Score", "Buffa Hypoxia Score", "MSI MANTIS Score", "MSIsensor Score", "Ragnum Hypoxia Score", "Winter Hypoxia Score") in `data_score`; the treatment and 'Overall Survival (days)' in `data_diag`, the stage of the cancer in `data_red`? ; etc

Meaning of the Biological scores:

-   **Aneuploidy Score** -\> measures the level of chromosome instability in a tumor, how many chromosomes or chromosome segments have abnormal copy numbers -\> correlates with tumor aggressiveness and poor prognosis -\> relate with stages (HOPEFULLY more advanced stages = more aneuploidy score)

    ```{r}
    #| fig.width=10
    #| fig.height=6

    # Relate Cancer Stage with Aneuploidy Score
    plt_aneuploidy_stage <- data_code |>
      full_join(data_score, by = "Patient ID") |> 
      rename(`Cancer Stage` = `Neoplasm Disease Stage American Joint Committee on Cancer Code`) |>
      ggplot(aes(x = `Cancer Stage`, 
                 y = `Aneuploidy Score`, 
                 fill = `Cancer Stage`)) +
      geom_boxplot() +
      theme_minimal() +
      ggtitle("Aneuploidy Score by Cancer Stage") +
      theme(axis.text.x = element_text(angle = 45, hjust = 1))

    plt_aneuploidy_stage
    ```

-   **Buffa Hypoxia Score** -\> Tumor hypoxia (low oxygen levels) based on a gene expression - note: Hypoxic tumors tend to be more resistant to radiotherapy (test that) -\> same for Ragnum Hypoxia Score; Winter Hypoxia Score

    ```{r}
    #| fig.width=20
    #| fig.height=50

    # Hypoxia Data  
    data_hypoxia <- data_full |>
      full_join(data_score, by = "Patient ID")

    data_hypoxia <- data_hypoxia |>
      select(PatientID = `Patient ID`,
             `Buffa Hypoxia Score`,
             `Ragnum Hypoxia Score`,
             `Winter Hypoxia Score`,
             `Radiation Therapy`,
             `Overall Survival (days)`,
             `Disease-specific Survival status (des)`,
              `Tumor Type`) |>
      filter(!is.na(`Radiation Therapy`))


    ggplot(data_hypoxia, aes(x = `Radiation Therapy`,
                             y = `Buffa Hypoxia Score`,
                             fill = `Radiation Therapy`)) +
      geom_boxplot(alpha = 0.6) +
      geom_jitter(width = 0.2,
                  alpha = 0.5) +
      theme_minimal() +
      labs(title = "Buffa Hypoxia Score vs Radiation Therapy")

    # Hypoxia Groups (low vs high) 
    hyp_med <- median(data_hypoxia[["Buffa Hypoxia Score"]], na.rm = TRUE)

    data_hypoxia <- data_hypoxia |>
      mutate(Hypoxia_Group = case_when(
          `Buffa Hypoxia Score` > hyp_med ~ "High Hypoxia Score",
          `Buffa Hypoxia Score` <= hyp_med ~ "Low Hypoxia Score"
        ))

     
    # Assign values to the status: 1 = 'dead with tumor' and 0 = 'alive or dead tumor free'
    data_hypoxia <- data_hypoxia |> 
      mutate(Status_Tumor = case_when(
          `Disease-specific Survival status (des)` == "DEAD WITH TUMOR" ~ 1,
          `Disease-specific Survival status (des)` == "ALIVE OR DEAD TUMOR FREE" ~ 0
        ))


    # Survival Plot: High vs Low Hypoxia in Radiation therapy

    # Filter only patients who received radiation therapy
    data_Radiation_Hyp <- data_hypoxia |>
      filter(`Radiation Therapy` == "Yes")

    #  survival 
    surv_Rad <- Surv(
      time = data_Radiation_Hyp$`Overall Survival (days)`,
      event = data_Radiation_Hyp$`Status_Tumor`)

    # KM model comparing High vs Low hypoxia 
    compare_Rad <- survfit(surv_Rad ~ Hypoxia_Group, data = data_Radiation_Hyp)

    # Plot
    ggsurvplot(
      compare_Rad,
      data = data_Radiation_Hyp,
      pval = TRUE,    # Take out if we just want to vizualize
      risk.table = FALSE,     # I think its cool but but we can take out
      palette = c("red", "blue"),
      legend.title = "Hypoxia Group",
      legend.labs = c("High Hypoxia", "Low Hypoxia"),
      title = "Survival of Radiation Therapy Patients\nHigh vs Low Hypoxia (Buffa Score)"
    )

    ```

    Everything again with the **Winter hypoxia score**

    ```{r}
    #| fig.width=20
    #| fig.height=50

    ggplot(data_hypoxia, aes(x = `Radiation Therapy`,
                             y = `Winter Hypoxia Score`,
                             fill = `Radiation Therapy`)) +
      geom_boxplot(alpha = 0.6) +
      geom_jitter(width = 0.2,
                  alpha = 0.5) +
      theme_minimal() +
      labs(title = "Winter Hypoxia Score vs Radiation Therapy")

    # Hypoxia Groups (low vs high) 
    hyp_med <- median(data_hypoxia[["Winter Hypoxia Score"]], na.rm = TRUE)

    data_hypoxia <- data_hypoxia |>
      mutate(Hypoxia_Group = case_when(
          `Winter Hypoxia Score` > hyp_med ~ "High Hypoxia Score",
          `Winter Hypoxia Score` <= hyp_med ~ "Low Hypoxia Score"
        ))

     
    # Assign values to the status: 1 = 'dead with tumor' and 0 = 'alive or dead tumor free'
    data_hypoxia <- data_hypoxia |> 
      mutate(Status_Tumor = case_when(
          `Disease-specific Survival status (des)` == "DEAD WITH TUMOR" ~ 1,
          `Disease-specific Survival status (des)` == "ALIVE OR DEAD TUMOR FREE" ~ 0
        ))


    # Survival Plot: High vs Low Hypoxia in Radiation therapy

    # Filter only patients who received radiation therapy
    data_Radiation_Hyp <- data_hypoxia |>
      filter(`Radiation Therapy` == "Yes")

    #  survival 
    surv_Rad <- Surv(
      time = data_Radiation_Hyp$`Overall Survival (days)`,
      event = data_Radiation_Hyp$`Status_Tumor`)

    # KM model comparing High  Low hypoxia 
    compare_Rad <- survfit(surv_Rad ~ Hypoxia_Group, data = data_Radiation_Hyp)

    # Plot
    ggsurvplot(
      compare_Rad,
      data = data_Radiation_Hyp,
      pval = TRUE,    # Take out if we just want to vizualize
      risk.table = FALSE,     # I think its cool but but we can take out
      palette = c("red", "blue"),
      legend.title = "Hypoxia Group",
      legend.labs = c("High Hypoxia", "Low Hypoxia"),
      title = "Survival of Radiation Therapy Patients\nHigh vs Low Hypoxia (Winter Score)"
    )


    ```

-   Everything again with **Ragnum Hypoxia Score**

    ```{r}
    #| fig.width=20
    #| fig.height=50

    ggplot(data_hypoxia, aes(x = `Radiation Therapy`,
                             y = `Ragnum Hypoxia Score`,
                             fill = `Radiation Therapy`)) +
      geom_boxplot(alpha = 0.6) +
      geom_jitter(width = 0.2,
                  alpha = 0.5) +
      theme_minimal() +
      labs(title = "Ragnum Hypoxia Score vs Radiation Therapy")

    # Hypoxia Groups (low vs high) 
    hyp_med <- median(data_hypoxia[["Ragnum Hypoxia Score"]], na.rm = TRUE)

    data_hypoxia <- data_hypoxia |>
      mutate(Hypoxia_Group = case_when(
          `Ragnum Hypoxia Score` > hyp_med ~ "High Hypoxia Score",
          `Ragnum Hypoxia Score` <= hyp_med ~ "Low Hypoxia Score"
        ))

     
    # Assign values to the status: 1 = 'dead with tumor' and 0 = 'alive or dead tumor free'
    data_hypoxia <- data_hypoxia |> 
      mutate(Status_Tumor = case_when(
          `Disease-specific Survival status (des)` == "DEAD WITH TUMOR" ~ 1,
          `Disease-specific Survival status (des)` == "ALIVE OR DEAD TUMOR FREE" ~ 0
        ))


    # Survival Plot: High vs Low Hypoxia in Radiation therapy

    # Filter only patients who received radiation therapy
    data_Radiation_Hyp <- data_hypoxia |>
      filter(`Radiation Therapy` == "Yes")

    #  survival 
    surv_Rad <- Surv(
      time = data_Radiation_Hyp$`Overall Survival (days)`,
      event = data_Radiation_Hyp$`Status_Tumor`)

    # KM model comparing High  Low hypoxia 
    compare_Rad <- survfit(surv_Rad ~ Hypoxia_Group, data = data_Radiation_Hyp)

    # Plot
    ggsurvplot(
      compare_Rad,
      data = data_Radiation_Hyp,
      pval = TRUE,    # Take out if we just want to vizualize
      risk.table = FALSE,     # I think its cool but but we can take out
      palette = c("red", "blue"),
      legend.title = "Hypoxia Group",
      legend.labs = c("High Hypoxia", "Low Hypoxia"),
      title = "Survival of Radiation Therapy Patients\nHigh vs Low Hypoxia (Ragnum Score)"
    )
    ```

-   **MSI MANTIS Score** -\> measures microsatellite instability (MSI), a condition where repetitive DNA sequences (microsatellites) are prone to errors due to defective DNA mismatch repair

-   MSIsensor Score -\> same as MSI MANTIS Score, but in percentage basically

**Overall correlation between tumor metrics (genomic instability vs biological scores) -\>\> in description??**

```{r}
#| fig.width=10
#| fig.height=6

library(ggplot2)
library(corrplot)

data_full <- data_gene |>
  full_join(data_score, by = "Patient ID") |>
  full_join(data_diag, by = "Patient ID") |>
  full_join(data_metrics, by = "Patient ID")

# rename things = easier
data_full <- data_full |>
  rename(
    TMB = `TMB (nonsynonymous)`,
    MutCount = `Mutation Count`,
    FGA = `Fraction Genome Altered`,
    TBL = `Tumor Break Load`
  )

# Select  columns of interest
cols_to_compare <- c("TMB", "MutCount", "FGA", "TBL",
                     "Aneuploidy Score", "Buffa Hypoxia Score",
                     "MSI MANTIS Score", "MSIsensor Score",
                     "Ragnum Hypoxia Score", "Winter Hypoxia Score")

data_numeric <- data_full |> 
  select(all_of(cols_to_compare))

correl_matrix <- cor(data_numeric |>
                    select(cols_to_compare), 
                    use = "complete.obs")

corrplot(correl_matrix, method = "color",tl.col = "black")


```
