---
title: "06_analysis_2"
format:
  html:
    embed-resources: true
editor: visual
---

## **Loading libraries**

```{r}
library("patchwork")
library("tidyverse")
library("ggplot2")
library("corrplot")
library("here")
```

## **Loading data**

Loading the data from the previous augmentation step *03_augment*.

```{r}
#| message: false
data_metrics <- read_tsv(here("data","data_metrics_aug.tsv"))
data_gene <- read_tsv(here("data","data_gene_aug.tsv"))
data_score <- read_tsv(here("data","data_score_aug.tsv"))
data_code <- read_tsv(here("data","data_code_aug.tsv"))
data_diag <- read_tsv(here("data","data_diag_aug.tsv"))
```

## Data Description/ Notes for Analysis

Summary of the data:

-   **data_metrics** — patient demographics (age, sex, race, ancestry)
-   **data_gene** — genomic instability metrics (mutation count, TMB, fraction genome altered).
-   **data_score** — biological scores (aneuploidy, hypoxia, MSI).
-   **data_code** — cancer staging and clinical classification codes.
-   **data_diag** — diagnoses, survival outcomes, therapy history.

## Genomic Alterations & Instability Analysis (**data_gene)**

### Relationship between continuous genomic instability measures and overall survival (days)

```{r}
#| fig-width: 15
#| fig-height: 7

# Joining tables and filter out outliers
data_gene_diag <- full_join(data_gene, data_diag, by = "Patient ID")

# Reshape to Long Dataset
data_long <- data_gene_diag |>
  mutate(
    TMB = `TMB (nonsynonymous)`,
    MutCount = `Mutation Count`,
    FracGenome = `Fraction Genome Altered`,
    TumorLoad = `Tumor Break Load`
  ) |>
  pivot_longer(
    cols = c(TMB, MutCount, FracGenome, TumorLoad),
    names_to = "Metric",
    values_to = "Value"
  )

# Compute medians per metric
medians <- data_long |>
  group_by(Metric) |>
  summarize(med = median(Value, na.rm = TRUE))

data_long <- data_long |>
  left_join(medians, by = "Metric") |>
  mutate(
    Group = case_when(
      Value > med ~ "Higher Genomic Instability",
      Value <= med ~ "Lower Genomic Instability"
      ),
    Metric = factor(
      Metric,
      labels = c(
        "TMB (nonsynonymous)",
        "Mutation Count",
        "Fraction Genome Altered",
        "Tumor Break Load"
      )))

## PLOTTING
# Scatter/LOESS Plot
plt_scatter <- ggplot(
  data_long,
  aes(x = `Overall Survival (days)`, 
      y = Value, 
      color = Group)
) +
  geom_point(alpha = 0.6) +
  geom_smooth(method = "loess", color = "blue") +
  facet_wrap(~ Metric, scales = "free_y") +
  theme_minimal() +
  theme(
    plot.title = element_text(hjust = 0.5, face = "bold")
  ) +
  labs(title = "Overall Survival vs Genomic Metrics")

# Present Plots
plt_scatter
```

Survival distributions indicate that higher genomic instability, particularly for Mutation Count, Fraction Genome Altered, and Tumor Break Load, is generally associated with slightly shorter survival, although the limited sample size in the longer-survival range may affect this The same analysis suggests that for TMB, the very highest values may correspond to longer survival, although the same sample size limitation is present.

As shown by both the scatter plot and descriptive statistics, Mutation Count and TMB (nonsynonymous) are highly correlated, with very similar distributions. Therefore, from this point forward only Mutation Count was visualized, since plotting both variables would not provide any additional insight.

### **Tumor types and Genomic differences across them (not meaningful for further analysis)**

-   **Context**: This section compares genomic instability metrics (Mutation Count, TMB, Fraction Genome Altered, Tumor Break Load) across different tumor types, to determine the existance of big difference between them.

-   **Analysis from Plots**: The 'Ductal Adenocarcinoma' appears to have lower fraction of the genome altered and a slight higher tumor break load compared to others subtypes, altough the differences are very discrete. The other tumor types were discarded due to lack of data.

```{r}
#| fig-width: 10
#| fig-height: 5

data_gene_diag |>
  filter(
    `Tumor Type` %in% c(
      "Pancreas Adenocarcinoma, Ductal Type",
      "Pancreas Adenocarcinoma, Other Subtype"
    )
  ) |>
  pivot_longer(
    cols = c(
      `Mutation Count`,
      `Fraction Genome Altered`,
      `Tumor Break Load`
    ),
    names_to = "Metric",
    values_to = "Value"
  ) |>
  ggplot(aes(x = `Tumor Type`, y = Value, fill = `Tumor Type`)) +
  geom_boxplot() +
  facet_wrap(~ Metric, nrow = 1, scales = "free_y") +
  theme_minimal() +
  theme(
    axis.title.x = element_blank(),
    axis.text.x  = element_blank(),
    axis.ticks.x = element_blank()
  )
```

### Distribution of Cancer Stages and Genomic Instability

-   **Context**: Cancer stage reflects the extent of tumor progression, while genomic instability measures (Mutation Count, Fraction Genome Altered, and Tumor Break Load) capture different aspects of chromosomal and genetic instability. These features are often associated with tumor aggressiveness and poor prognosis. This set of plots explores how cancer stage distributions relate to genomic instability across pancreatic adenocarcinoma subtypes (Ductal Type vs Other Subtypes).

-   The stage distribution plot in describe shows a concentration of patients in **Stage IIB**, indicating the dataset is heavily skewed toward mid-stage disease.

```{r}
data_gene_code = full_join(data_gene, data_code, by = "Patient ID")
```

```{r}
# 1) Find stages with > 10 samples
stages_keep <- data_gene_code |>
  rename(NDSA_stage = `Neoplasm Disease Stage American Joint Committee on Cancer Code`) |>
  count(NDSA_stage, name = "n_samples") |>
  filter(n_samples > 10)

# 2) Labels
stage_labels <- stages_keep |>
  mutate(stage_label = paste0(NDSA_stage, " (n=", n_samples, ")")) |>
  select(NDSA_stage, stage_label)

# 2) Keep full data only for those stages
data_gene_code_filtered <- data_gene_code |>
  rename(NDSA_stage = `Neoplasm Disease Stage American Joint Committee on Cancer Code`) |>
  filter(!is.na(NDSA_stage)) |> 
  semi_join(stages_keep, by = "NDSA_stage") 
```

```{r}
#| fig-width: 10
#| fig-height: 5
data_gene_code_filtered |>
  left_join(stage_labels, by = "NDSA_stage") |>
  pivot_longer(
    cols = c(
      `Fraction Genome Altered`,
      `Mutation Count`,
      `Tumor Break Load`
    ),
    names_to = "Genomic_Metric",
    values_to = "Value"
  ) |>
  ggplot(aes(x = stage_label, y = Value)) +
  geom_violin() +
  geom_boxplot(width = 0.2) +
  facet_wrap(~ Genomic_Metric, 
             scales = "free_y", 
             nrow = 1) +
  theme(axis.text.x = element_text(angle = 45, 
                                   hjust = 1)) +
  labs(
    x = "AJCC stage (n per stage)",
    y = "Genomic metric")
```

To begin, we only included tumor classes with at least 10 patients per AJCC stage, as smaller groups would not provide a reliable representation of the underlying distribution.

-   There’s considerable overlap between neighboring stages, so stage alone doesn’t perfectly separate genomic burden.

-   One could say that Tumor Break Load tends to increase with tumor stage, as indicated by higher medians and a wider spread of values in the boxplots for advanced stages.

-   **TBL measures:** it is the sum of unbalanced structural-variant breakpoints, a quantitative readout of **Structural Variants driven chromosomal instability**.

-   So, *in theory*, as tumors accumulate Structural Variants and copy‑number chaos over time, **TBL should rise**, and high TBL is often seen in more aggressive, later‑stage or poorer‑prognosis disease.

### Heatmap of the pairwise correlations between genomic instability and hypoxia-related metrics

```{r}
#| fig-width: 10
#| fig-height: 6

data_full <- data_gene |>
  full_join(data_score, by = "Patient ID") |>
  full_join(data_diag, by = "Patient ID") |>
  full_join(data_metrics, by = "Patient ID")

# Select  columns of interest
cols_to_compare <- c("TMB (nonsynonymous)", "Mutation Count", 
                     "Fraction Genome Altered", "Tumor Break Load",
                     "Aneuploidy Score", "Buffa Hypoxia Score",
                     "MSI MANTIS Score", "MSIsensor Score",
                     "Ragnum Hypoxia Score", "Winter Hypoxia Score")

data_numeric <- data_full |> 
  select(all_of(cols_to_compare))

correl_matrix <- cor(data_numeric |>
                    select(cols_to_compare), 
                    use = "complete.obs")

corrplot(correl_matrix, method = "color",tl.col = "black")

```

This plot was used to assess relationships between genomic instability metrics and hypoxia scores. Focusing on the first four columns (genomic instability measures) and the last six rows (hypoxia scores), no strong blue stand out, indicating that hypoxia scores are not strongly correlated with the genomic instability variables in this dataset.

## Biological scores Analysis (score_data)

### Distribution of Aneuploidy Scores across different Cancer Stages

```{r}
#| fig-width: 10
#| fig-height: 6

# Relate Cancer Stage with Aneuploidy Score
plt_aneuploidy_stage <- data_code |>
  full_join(data_score, by = "Patient ID") |> 
  rename(`Cancer Stage` = `Neoplasm Disease Stage American Joint Committee on Cancer Code`) |>
  drop_na("Cancer Stage") |>
  ggplot(aes(x = `Cancer Stage`, 
             y = `Aneuploidy Score`, 
             fill = `Cancer Stage`)) +
  geom_boxplot() +
  theme_minimal() +
  theme(
    legend.position = "none",
    plot.title = element_text(hjust = 0.5, face = "bold")
  ) +
  ggtitle("Aneuploidy Score by Cancer Stage") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

plt_aneuploidy_stage
```

The Aneuploidy Score, which quantifies chromosomal instability and is often linked to tumor aggressiveness and poor prognosis, was analyzed to assess whether higher cancer stages correspond to increased chromosomal abnormalities. The plot shows no clear increase in Aneuploidy Score with advancing stage, which may reflect the limited stage diversity in the dataset, particularly the small number of Stage I cases.

```{r}
metric_for_order <- "Winter Hypoxia Score"

df_long <- data_score |>
  select(`Patient ID`,
         `MSI MANTIS Score`,
         `MSIsensor Score`,
         `Buffa Hypoxia Score`,
         `Ragnum Hypoxia Score`,
         `Winter Hypoxia Score`) |>
  drop_na() |>
  pivot_longer(
    cols = -`Patient ID`,
    names_to = "Metric",
    values_to = "Score"
  ) |>
  group_by(Metric) |>
  mutate(
    Score_scaled = (Score - mean(Score, na.rm = TRUE)) /
                   sd(Score, na.rm = TRUE)
  ) |>
  ungroup() |>
  group_by(`Patient ID`) |>
  mutate(
    order_value = Score_scaled[Metric == metric_for_order][1]
  ) |>
  ungroup() |>
  mutate(
    `Patient ID` = fct_reorder(`Patient ID`, order_value, .desc = TRUE)
  )

ggplot(df_long,
       aes(x = `Patient ID`, y = Metric, fill = Score_scaled)) +
  geom_tile() +
  scale_fill_viridis_c() +
  theme_minimal() +
  theme(
    axis.text.x = element_blank(),
    axis.ticks.x = element_blank()
  ) +
  labs(
    title = "MSI scores across patients (z-scaled)",
    x = "Patient", y = "Metric", fill = "z-score"
  )
```

## SURVIVAL 3

#### Distribution of Hypoxia Scores in Patients Receiving Radiation Therapy, and Comparison of overall survival (days) of patients treated with radiotherapy depending on levels of hypoxia. ----\> on survival_analysis

#### **Overall correlation between tumor metrics (genomic instability vs biological scores)**

-   **Context**: This section calculates and visualizes the pairwise correlation matrix between the four genomic instability metrics (TMB, Mutation Count, FGA, TBL) and the six biological scores (Aneuploidy Score, Buffa Hypoxia Score, MSI MANTIS Score, MSIsensor Score, Ragnum Hypoxia Score, Winter Hypoxia Score).

-   **Analysis from Plot:**

    -   **Genomic Instability Correlation:** TMB and Mutation Count are highly positively correlated, as both measure the overall burden of small-scale point mutations. Fraction Genome Altered and Tumor Break Load, which measure large-scale chromosomal instability, show a slightly more negative correlation with TMB/Mutation Count. This suggests these two forms of genomic instability may be somewhat distinct in this tumor cohort.

    -   **Hypoxia Score Correlation:** The three Hypoxia Scores (Buffa, Ragnum, Winter) show a positive correlation, confirming they all effectively measure the same underlying stressor: tumor hypoxia (low oxygen). Hypoxia Scores also show a positive correlation with Tumor Break Load (TBL), suggesting that the low-oxygen environment may contribute to large-scale structural DNA damage.

    -   The two MSI metrics (**MSI MANTIS** and **MSIsensor**) show almost no correlation, which is unexpected given they track the same defective DNA mismatch repair mechanism. The MSI MANTIS Score shows a slight positive correlation with Mutation Count and TMB. This is biologically plausible, as defective mismatch repair (causing MSI) simultaneously increases the rate of overall point mutations (TMB/Mutation Count).

    -   **Aneuploidy:** Aneuploidy Score and Fraction of Genome Altered (FGA) show a moderate positive correlation. This is expected, as both are direct measures of chromosomal instability.
