---
  title: "04_describe"
format: 
  html:
  embed-resources: true
editor: visual
---

## **Loading libraries**

```{r}
library("tidyverse")
library("here")
```

## **Loading data**

Loading the data from the previous augmentation step *03_augment*.

```{r}
data_metrics <- read_tsv(here("data","data_metrics_aug.tsv"))
data_gene <- read_tsv(here("data","data_gene_aug.tsv"))
data_score <- read_tsv(here("data","data_score_aug.tsv"))
data_code <- read_tsv(here("data","data_code_aug.tsv"))
data_diag <- read_tsv(here("data","data_diag_aug.tsv"))
```

## Metrics summary

List the metrics, the number of metrics, the types within the metrics, and the median values within each types.

Males exhibit a higher mean age at diagnosis compared to females. Age groups of \[30,40\] and \[80,90\] contain too few individuals to support reliable statistical analysis.​

Race, Genetic Ancestry Label, and Ethnicity Category strongly overlap in our data, with most individuals of European ancestry also classified as White and similar by ethnicity. This high correlation limits diversity across subgroups, rendering analyses stratified by these categories unreliable for drawing meaningful conclusions.

```{r}
str(data_metrics)

metric_cols <- c(
  "Diagnosis Age",
  "Sex",
  "Genetic Ancestry Label",
  "Race Category",
  "Ethnicity Category"
)

metrics_summary <-
  data_metrics |>
  
  select(all_of(metric_cols)) |>
  mutate(across(everything(), as.character)) |>
  pivot_longer(
    cols = everything(),
    names_to = "Columns",
    values_to = "Value"
  ) |>
  
  filter(!is.na(Value)) |>
  group_by(Columns) |>

  summarise(
    n_distinct = n_distinct(Value),
    Types = tibble(Value = Value) |>
      count(Value) |>
      arrange(desc(n)) |>
      mutate(type_str = str_c(Value, " (", n, ")")) |>
      pull(type_str) |>
      str_c(collapse = ", "),
    .groups = "drop"
  )

metrics_summary

patient_row <-
  data_metrics |>
  summarise(
    Columns = "Patient ID",
    n_distinct = n_distinct(`Patient ID`, na.rm = TRUE),
    Types = str_c("Total entries: ", n()),
    .groups = "drop"
  )

patient_row

p_value <- data_metrics |>
  summarise(p = wilcox.test(`Diagnosis Age` ~ Sex)$p.value) |>
  pull(p) |>
  signif(1)

ggplot(data = data_metrics, (aes(x=Sex, y=`Diagnosis Age`))) +
  geom_violin(aes(fill = Sex), color = NA, alpha = 0.5) +
  stat_summary(fun = median, geom = "crossbar", color = "red", size = 1) +
  geom_jitter(position = position_jitter(width = 0.3, seed = 123)) +
  annotate(
    "text",
    x = 1.5,
    y = max(data_metrics$`Diagnosis Age`, na.rm = TRUE),
    label = paste("p =", p_value),
    size = 6) +
  theme(
        panel.background = element_blank(),
        axis.text = element_text(size = 16),
        axis.title = element_text(size = 16),
        legend.text = element_text(size = 16),
        legend.title = element_text(size = 16))

data_metrics |>
  count(Sex, `Genetic Ancestry Label`, `Race Category`, `Ethnicity Category`) |>
  ggplot(aes(x = `Genetic Ancestry Label`, y = `Race Category`, fill = n)) +
  geom_tile(color = "white") +
  geom_text(aes(label = n), color = "black", size = 3) +
  facet_wrap(Sex ~ `Ethnicity Category`) +
  scale_fill_viridis_c(direction = -1)

```

## **Gene summary**

Providing core statistics overview of the genetic mutation data with a basic descriptive stats table and histograms and box plots. Outliers are removed to avoid making the graphs unreadable.

```{r}
str(data_gene)

gene_cols <- c(
  "Fraction Genome Altered",
  "Mutation Count",
  "TMB (nonsynonymous)",
  "Tumor Break Load"
)

gene_long <- data_gene |>
  select(all_of(gene_cols)) |>
  pivot_longer(
    cols = everything(),
    names_to = "Category",
    values_to = "Value"
  )

gene_no_ext <- gene_long |>
  group_by(Category) |>
  filter(Value != max(Value, na.rm = TRUE)) |> 
  ungroup()

plot_gene <- function(df, gene_title) {

  p <- df |>
    ggplot(aes(x = Category, y = Value, fill = Category)) +
    geom_violin(alpha = 0.35, color = "gray40") +
    geom_jitter(width = 0.1, alpha = 0.4, color = "black", size = 2) +
    labs(title = paste("Distribution of", gene_title), x = NULL, y = gene_title) +
    theme_minimal(base_size = 14) +
    theme(legend.position = "none")

  print(p)
}

gene_no_ext |>
  group_by(Category) |>
  group_walk(~ plot_gene(mutate(.x, Category = .y$Category), .y$Category))

gene_summary <- gene_no_ext |>
  group_by(Category) |>
  summarise(
    Mean = round(mean(Value, na.rm = TRUE), 2),
    Median = round(median(Value, na.rm = TRUE), 2),
    Standard_Deviation = round(sd(Value, na.rm = TRUE), 2),
    .groups = "drop"
  )

gene_summary
```

## Score Summary

Providing core statistics overview of the cancer scoring data with a basic descriptive stats table and histograms and box plots. Outliers not removed/present.

```{r}
str(data_score)

score_cols_no_hypoxia <- c(
  "Aneuploidy Score",
  "MSI MANTIS Score",
  "MSIsensor Score",
  "Buffa Hypoxia Score",
  "Ragnum Hypoxia Score",
  "Winter Hypoxia Score"
)

score_long <- data_score |>
  select(all_of(score_cols_no_hypoxia)) |>
  mutate(across(everything(), as.numeric)) |>
  pivot_longer(
    cols = everything(),
    names_to = "Score",
    values_to = "Value"
  )

score_no_ext <- score_long |>
  group_by(Score) |>
  filter(Value != max(Value, na.rm = TRUE)) |>
  ungroup()

plot_score <- function(df, score_title) {

  p <- df |>
    ggplot(aes(x = Score, y = Value, fill = Score)) +
    geom_violin(alpha = 0.35, color = "gray40") +
    geom_jitter(width = 0.1, alpha = 0.4, color = "black", size = 2) +
    labs(title = paste("Distribution of", score_title), x = NULL, y = score_title) +
    theme_minimal(base_size = 14) +
    theme(legend.position = "none")

  print(p)
}

hypoxia_scores <- c(
  "Buffa Hypoxia Score",
  "Ragnum Hypoxia Score",
  "Winter Hypoxia Score"
)

hypoxia_long <- score_no_ext |>
  filter(Score %in% hypoxia_scores)

  p_hypoxia <- hypoxia_long |>
    ggplot(aes(x = Score, y = Value, fill = Score)) +
    geom_violin(alpha = 0.35, color = "gray40") +
    geom_jitter(width = 0.1, alpha = 0.4, color = "black", size = 1.8) +
    facet_wrap(~ Score, ncol = 3) +
    labs(x = NULL, y = "Value") +
    theme_minimal(base_size = 14) +
    theme(
      legend.position = "none",
      strip.text = element_text(face = "bold"),
      axis.text.x = element_blank(),      # remove tick text
      axis.ticks.x = element_blank()      # optional: remove tick marks
    )

p_hypoxia

score_no_ext |>
  group_by(Score) |>
  group_walk(~ plot_score(mutate(.x, Score = .y$Score), .y$Score))

score_summary <- score_no_ext |>
  group_by(Score) |>
  summarise(
    Mean = round(mean(Value, na.rm = TRUE), 2),
    Median = round(median(Value, na.rm = TRUE), 2),
    Standard_Deviation = round(sd(Value, na.rm = TRUE), 2),
    .groups = "drop"
  )

score_summary
```

## Codes Summary

Summary of the various types of cancer codes.

```{r}
str(data_code)

make_types <- function(dataset_category) {
  dataset_category |>
    na.omit() |>
    as_tibble() |>
    count(value = value, sort = TRUE) |>
    mutate(type = paste0(value, " (", n, ")")) |>
    pull(type) |>
    paste(collapse = ", ")
}

neoplasm_stage_types <- make_types(data_code$`Neoplasm Disease Stage American Joint Committee on Cancer Code`)
icd_histology_types <- make_types(data_code$`International Classification of Diseases for Oncology, Third Edition ICD-O-3 Histology Code`)
icd_site_types <- make_types(data_code$`International Classification of Diseases for Oncology, Third Edition ICD-O-3 Site Code`)
joint_stage_types <- make_types(data_code$`American Joint Committee on Cancer Metastasis Stage Code`)
lymph_stage_types <- make_types(data_code$`Neoplasm Disease Lymph Node Stage American Joint Committee on Cancer Code`)
tumor_stage_types <- make_types(data_code$`American Joint Committee on Cancer Tumor Stage Code`)

neoplasm_stage_distinct <- data_code$`Neoplasm Disease Stage American Joint Committee on Cancer Code` |> na.omit() |> n_distinct()
icd_histology_distinct <- data_code$`International Classification of Diseases for Oncology, Third Edition ICD-O-3 Histology Code` |> na.omit() |> n_distinct()
icd_site_distinct <- data_code$`International Classification of Diseases for Oncology, Third Edition ICD-O-3 Site Code` |> na.omit() |> n_distinct()
joint_stage_distinct <- data_code$`American Joint Committee on Cancer Metastasis Stage Code` |> na.omit() |> n_distinct()
lymph_stage_distinct <- data_code$`Neoplasm Disease Lymph Node Stage American Joint Committee on Cancer Code` |> na.omit() |> n_distinct()
tumor_stage_distinct <- data_code$`American Joint Committee on Cancer Tumor Stage Code` |> na.omit() |> n_distinct()

code_summary <- tibble(
  
  Columns = c(
    "Neoplasm Disease Stage Code",
    "AJCC Tumor Stage Code",
    "AJCC Metastasis Stage Code",
    "ICD-O-3 Histology Code",
    "ICD-O-3 Site Code",
    "AJCC Lymph Node Stage Code"
  ),

  Distinct = c(
    neoplasm_stage_distinct,
    tumor_stage_distinct,
    joint_stage_distinct,
    icd_histology_distinct,
    icd_site_distinct,
    lymph_stage_distinct
  ),

  Types = c(
    neoplasm_stage_types,
    tumor_stage_types,
    joint_stage_types,
    icd_histology_types,
    icd_site_types,
    lymph_stage_types
  )
)

code_summary

code_vars <- data_code |>
  select(
    `Neoplasm Stage` = `Neoplasm Disease Stage American Joint Committee on Cancer Code`,
    `AJCC Tumor` = `American Joint Committee on Cancer Tumor Stage Code`,
    `AJCC Metastasis` = `American Joint Committee on Cancer Metastasis Stage Code`,
    `ICDO3 Histology` = `International Classification of Diseases for Oncology, Third Edition ICD-O-3 Histology Code`,
    `ICDO3 Site` = `International Classification of Diseases for Oncology, Third Edition ICD-O-3 Site Code`,
    `AJCC Lymph_Node` = `Neoplasm Disease Lymph Node Stage American Joint Committee on Cancer Code`
  ) |>
  mutate(across(everything(), ~ as.numeric(as.factor(.x))))

cor_matrix <- cor(code_vars, use = "pairwise.complete.obs")
cor_df <- as.data.frame(cor_matrix) |>
  rownames_to_column("Var1") |>
  pivot_longer(-Var1, names_to = "Var2", values_to = "Correlation")

ggplot(cor_df, aes(Var1, Var2, fill = Correlation)) +
  geom_tile() +
  scale_fill_gradient2(low = "blue", mid = "white", high = "red") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  labs(title = "Cancer Code Correlation Heatmap", x = NULL, y = NULL)
```

## Diagnoses Summary

Summary of the different cancer diagnoses.

```{r}
str(data_diag)

#----- numerical data
diagnosis_list_numeric <- list(
  "Overall Survival (days)"        = data_diag$`Overall Survival (days)`,
  "Overall Survival (months)"      = data_diag$`Overall Survival (Months)`,
  "Disease Free (months)"          = data_diag$`Disease Free (Months)`,
  "Disease Free (days)"            = data_diag$`Disease Free (days)`,
  "Progression Free (months)"      = data_diag$`Progress Free Survival (Months)`,
  "Progression Free (days)"        = data_diag$`Progress Free Survival (days)`
)

plot_diagnosis <- function(values, title) {

  df <- tibble(Value = values)

  p1 <- ggplot(df, aes(Value)) +
    geom_histogram(bins = 30, fill = "skyblue", color = "white") +
    labs(title = paste("Distribution of", title), x = title, y = "Frequency") +
    theme_minimal()

  p2 <- ggplot(df, aes(y = Value)) +
    geom_boxplot(fill = "lightgray") +
    labs(title = paste("Boxplot of", title), y = title) +
    theme_minimal()

  print(p1)
  print(p2)
}

walk2(
  .x = diagnosis_list_numeric,
  .y = names(diagnosis_list_numeric),
  .f = ~ plot_diagnosis(.x, .y)
)

diagnosis_numerical_summary <- tibble(
  Name = names(diagnosis_list_numeric),
  Mean = map_dbl(diagnosis_list_numeric, ~ mean(.x, na.rm = TRUE)),
  Median = map_dbl(diagnosis_list_numeric, ~ median(.x, na.rm = TRUE)),
  Standard_Deviation = map_dbl(diagnosis_list_numeric, ~ sd(.x, na.rm = TRUE))
) |>
  mutate(across(-Name, ~ round(.x, 2)))

diagnosis_numerical_summary


#----- categorical data
sort_data <- function(vec) {
  tibble(value = vec) |>
    drop_na() |>
    count(value, sort = TRUE) |>
    mutate(out = paste0(value, " (", n, ")")) |>
    pull(out) |>
    paste(collapse = ", ")
}

diagnosis_list_categorical <- list(
  "Tumor Type"                             = data_diag$`Tumor Type`,
  "Overall Survival Status (bin)"          = data_diag$`Overall Survival Status (bin)`,
  "Overall Survival Status (des)"          = data_diag$`Overall Survival Status (des)`,
  "Disease-specific Survival Status (des)" = data_diag$`Disease-specific Survival status (des)`,
  "Disease Free Status (bin)"              = data_diag$`Disease Free Status (bin)`,
  "Disease Free Status (des)"              = data_diag$`Disease Free Status (des)`,
  "Progression Free Status (bin)"          = data_diag$`Progression Free Status (bin)`,
  "Progression Free Status (des)"          = data_diag$`Progression Free Status (des)`,
  "Prior Diagnosis"                        = data_diag$`Prior Diagnosis`,
  "Radiation Therapy"                      = data_diag$`Radiation Therapy`,
  "Primary Lymph Node Assessment"          = data_diag$`Primary Lymph Node Presentation Assessment`,
  "Person Neoplasm Status"                 = data_diag$`Person Neoplasm Cancer Status`,
  "New Neoplasm Event"                     = data_diag$`New Neoplasm Event Post Initial Therapy Indicator`,
  "Histologic Grade"                       = data_diag$`Neoplasm Histologic Grade`,
  "ICD-10 Classification"                  = data_diag$`ICD-10 Classification`
)

diagnosis_categorical_summary <- tibble(
  Name = names(diagnosis_list_categorical),
  n_distinct = map_int(diagnosis_list_categorical, ~ n_distinct(na.omit(.x))),
  Types = map_chr(diagnosis_list_categorical, sort_data)
)

diagnosis_categorical_summary
```

## Comparative Analysis of Variables Within a Common Theme

In our dataset (divided ones), multiple variables often classify similar attributes or quantify related parameters, so it is essential to analyze the correlations between these variables to uncover their relationships and potential interdependencies.

Furthermore, some variables may not be meaningful for analysis. For example, nominal variables that overwhelmingly dominate the dataset, where one category accounts for the vast majority of observations, may offer limited insight and can be excluded from detailed examination.

### Data metrics (data_metrics)

```{r}
counts <- data_metrics |> 
  group_by(Sex) |> 
  summarise(n = n())

lab_female <- paste0("Women (n = ", counts$n[counts$Sex == "Female"], ")")
lab_male   <- paste0("Men (n = ", counts$n[counts$Sex == "Male"], ")")

data_metrics |>
  ggplot(aes(x = `Diagnosis Age`, fill = Sex)) +
  geom_density(alpha = 0.5) +
  scale_fill_discrete(
    name   = "Sex",
    labels = c("Female" = lab_female,
               "Male"   = lab_male)
  )
```

### Genomic Alterations & Instability (data_gene) 

Fraction Genome Altered - The proportion of the tumor’s genome with copy number gains or losses, reflecting the extent of chromosomal instability present in the cancer cells.

Mutation Count - The total number of mutations identified in the tumor genome, typically including single-nucleotide variants and small insertions/deletions.

TMB (nonsynonymous) - Tumor Mutation Burden (TMB) is the number of nonsynonymous mutations (those that change protein sequences) per megabase of DNA in the tumor, and is used to estimate overall mutational load.

Tumor Break Load - The quantity or extent of structural variants in the tumor genome, representing breaks or major rearrangements, which indicate higher genomic instability in the tumor.

```{r}
library(corrplot)
cor_matrix <- cor(data_gene |>
                    select(`Fraction Genome Altered`, 
                           `Mutation Count`, 
                           `TMB (nonsynonymous)`, 
                           `Tumor Break Load`), 
                  use = "complete.obs")

corrplot(cor_matrix, method = "color")
```

We excluded one extreme outlier before plotting; the resulting positive correlation is consistent with random mutagenesis, where the proportion of nonsynonymous mutations remains relatively stable given a fixed coding sequence fraction.

### Cancer Classification Systems (data_code)

```{r}
ClassificationSystems <- c(
  "Neoplasm Disease Stage American Joint Committee on Cancer Code",
  "International Classification of Diseases for Oncology, Third Edition ICD-O-3 Histology Code",
  "International Classification of Diseases for Oncology, Third Edition ICD-O-3 Site Code",
  "American Joint Committee on Cancer Metastasis Stage Code",
  "Neoplasm Disease Lymph Node Stage American Joint Committee on Cancer Code",
  "American Joint Committee on Cancer Tumor Stage Code"
)
```

```{r}
class_sym_1 <- ClassificationSystems[1]  
class_sym_2 <- ClassificationSystems[3]

data_code |>
  count(!!sym(class_sym_1),
        !!sym(class_sym_2)) |>
  ggplot(aes(x = !!sym(class_sym_1),
             y = !!sym(class_sym_2),
             fill = n)) +
  geom_tile() +
  geom_text(aes(label = n), color = "white", size = 3) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  labs(fill = "Count")
```

It is also inherently difficult to evaluate “coherence” between these systems quantitatively, because they use different numbers of categories and different underlying metrics.

American Joint Committee on Cancer Metastasis Stage Code (MX) -\> Metastasis cannot be assessed or information is insufficient to assign an M category

Besides this always one class that has the greatest

```{r}
entropy_cat <- function(x) {
  p <- prop.table(table(x))
  -sum(p * log(p))
}
```

```{r}
vars_to_check <- c(
  ClassificationSystems[1], # Neoplasm disease stage
  ClassificationSystems[2], 
  ClassificationSystems[3],
  ClassificationSystems[4], # AJCC metastasis stage code
  ClassificationSystems[5], # Lymph node stage
  ClassificationSystems[6]  # Tumor stage
)

dispersion_tbl <-
  tibble(var_name = vars_to_check) |>
  mutate(
    entropy = map_dbl(var_name, ~ {
      x <- data_code[[.x]]
      x <- x[!is.na(x)]         # remove NAs for this variable
      entropy_cat(x)
    }),
    n_levels = map_int(var_name, ~ {
      x <- data_code[[.x]]
      n_distinct(x, na.rm = TRUE)
    })
  ) |>
  arrange(desc(entropy))

dispersion_tbl
```

The Neoplasm Disease Stage American Joint Committee on Cancer Code shows the highest entropy among the examined systems.

### Biological scores (aneuploidy, hypoxia, MSI) (**data_score)**

```{r}
data_score
```

```{r}
# Select only score columns
score_cols <- c(
  "Aneuploidy Score", 
  "Buffa Hypoxia Score",
  "MSI MANTIS Score", 
  "MSIsensor Score",
  "Ragnum Hypoxia Score", 
  "Winter Hypoxia Score"
)

correl_matrix <- data_score |>
  select(all_of(score_cols)) |>
  cor(use = "complete.obs")

corrplot(correl_matrix, 
         method = "color", 
         tl.col = "black")
```

-   Aneuploidy Score → chromosome-level copy-number changes.

-   MSI MANTIS / MSIsensor → microsatellite instability (DNA repair).

-   Buffa / Ragnum / Winter → hypoxia from expression, different gene signatures.

```{r}
data_score |> 
  ggplot(mapping = 
           aes( x = `Tissue Source Site Code`)) +
  geom_bar() +
  theme_minimal()
```

### Tumor Types distribution (**data_diag)**

```{r}
#| fig-height: 5

data_diag |>
  ggplot(aes(x = `Tumor Type`, 
             fill = `Tumor Type`)) +
  geom_bar() +
  theme(
    axis.title.x = element_blank(),
    axis.text.x  = element_blank(),
    axis.ticks.x = element_blank()
  )
```

Not enough disersity in tumor types.
